cmd/terraform.go

```diff
@@ -19,9 +23,29 @@ var terraformCmd = &cobra.Command{
 	Short:              "Execute Terraform commands (e.g., plan, apply, destroy) using Atmos stack configurations",
 	Long:               `This command allows you to execute Terraform commands, such as plan, apply, and destroy, using Atmos stack configurations for consistent infrastructure management.`,
 	FParseErrWhitelist: struct{ UnknownFlags bool }{UnknownFlags: true},
-	PostRunE: func(cmd *cobra.Command, args []string) error {
-		info := getConfigAndStacksInfo("terraform", cmd, args)
-		return hooks.RunE(cmd, args, &info)
+	PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+		// Get the config and stacks info
+		info := getConfigAndStacksInfo("terraform", cmd, append([]string{cmd.Name()}, args...))
+
+		// Initialize the CLI config
+		atmosConfig, err := cfg.InitCliConfig(info, true)
+		if err != nil {
+			return fmt.Errorf("error initializing CLI config: %w", err)
+		}
+
+		// Get the hooks
+		hooks, err := h.GetHooks(&atmosConfig, &info)
+		if err != nil {
+			return fmt.Errorf("error getting hooks: %w", err)
+		}
+
+		// Set the the context so it can be accessed by any chiild commands
+		ctx := context.WithValue(context.Background(), "atmosConfig", &atmosConfig)
```

Comment:
Atmos config is used by all commands except some `about`, `version` commands.
According to me we should add this at the root level.